File: src/python/student/student.py
class Student:
    """
    A class to represent a student in a system.
    """
    
    def __init__(self, student_id: int, name: str, age: int, major: str):
        """
        Initialize a student with ID, name, age, and major.
        """
        self.student_id = student_id
        self.name = name
        self.age = age
        self.major = major
        self.grades = {}
    
    def add_grade(self, course: str, grade: float):
        """
        Add a grade for a specific course.
        """
        self.grades[course] = grade
    
    def get_gpa(self) -> float:
        """
        Calculate and return the student's GPA.
        """
        if not self.grades:
            return 0.0
        return sum(self.grades.values()) / len(self.grades)
    
    def change_major(self, new_major: str):
        """
        Change the student's major.
        """
        self.major = new_major
    
    def __str__(self):
        """
        Return a string representation of the student.
        """
        return f"Student {self.name} (ID: {self.student_id}), Age: {self.age}, Major: {self.major}, GPA: {self.get_gpa():.2f}"

File: src/java/Student/Student.java
package com.mycompany.app;

import java.util.HashMap;
import java.util.Map;

/**
 * A class to represent a student in a system.
 */
public class Student {
    private int studentId;
    private String name;
    private int age;
    private String major;
    private Map<String, Double> grades;

    /**
     * Constructor to initialize a student with ID, name, age, and major.
     */
    public Student(int studentId, String name, int age, String major) {
        this.studentId = studentId;
        this.name = name;
        this.age = age;
        this.major = major;
        this.grades = new HashMap<>();
    }

    /**
     * Adds a grade for a specific course.
     */
    public void addGrade(String course, double grade) {
        grades.put(course, grade);
    }

    /**
     * Calculates and returns the student's GPA.
     */
    public double getGPA() {
        if (grades.isEmpty()) {
            return 0.0;
        }
        double total = 0.0;
        for (double grade : grades.values()) {
            total += grade;
        }
        return total / grades.size();
    }

    /**
     * Changes the student's major.
     */
    public void changeMajor(String newMajor) {
        this.major = newMajor;
    }

    @Override
    public String toString() {
        return String.format("Student %s (ID: %d), Age: %d, Major: %s, GPA: %.2f",
                name, studentId, age, major, getGPA());
    }
}
